Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -1435,3 +1435,118 @@ extension HTTPMethod: RawRepresentable {
         }
     }
 }
+
+extension HTTPResponseHead {
+    public var hasContentLength: Bool {
+        return headers.hasContentLength
+    }
+
+    public var contentLength: Int? {
+        return headers.contentLength
+    }
+}
+
+extension HTTPRequestHead {
+    // Returns {@code true} if and only if the specified message contains an expect header and the only expectation
+    // present is the 100-continue expectation. Note that this method returns {@code false} if the expect header is
+    // not valid for the message (e.g., the version on the message is HTTP/1.0).
+    public var isContinueExpected: Bool {
+        return headers.isContinueExpected(version: version)
+    }
+
+    // Returns {@code true} if the specified message contains an expect header specifying an expectation that is not
+    // supported. Note that this method returns {@code false} if the expect header is not valid for the message
+    // (e.g., the version on the message is HTTP/1.0).
+    public var isUnsupportedExpectation: Bool {
+        return headers.isUnsupportedExpectation(version: version)
+    }
+
+    public var hasContentLength: Bool {
+        return headers.hasContentLength
+    }
+
+    public var contentLength: Int? {
+        return headers.contentLength
+    }
+}
+
+extension HTTPHeaders {
+    var hasContentLength: Bool {
+        return self.contains(name: "content-length") && self[canonicalForm: "content-length"].count == 1
```

Comment:
This loops over the headers twice. We can remove `contains` and just use the `canonicalForm` check, as it's redundant with the previous one.